### ISSUE TITLE
Uncaught TypeError: Cannot read property '1' of null

### ISSUE THREAD

**Issue by piot** [davidpaulrosser/atom-glsl-preview#30]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.19.4 x64
**Electron**: 1.6.9
**OS**: Mac OS X 10.12.6
**Thrown From**: [glsl-preview](https://github.com/fordhurley/atom-glsl-preview) package 1.2.0

### Stack Trace

Uncaught TypeError: Cannot read property '1' of null

```
At /Users/Peter/.atom/packages/glsl-preview/lib/glsl-preview-view.js:16

TypeError: Cannot read property '1' of null
    at parseLineNumberFromErrorMsg (/packages/glsl-preview/lib/glsl-preview-view.js:16:12)
    at setTimeout (/packages/glsl-preview/lib/glsl-preview-view.js:165:29)
```

### Commands

```
     -2:50.3.0 core:save (input.hidden-input)
     -2:25.1.0 core:move-up (input.hidden-input)
     -2:24.8.0 editor:move-to-end-of-screen-line (input.hidden-input)
     -2:24.6.0 editor:newline (input.hidden-input)
     -2:12.8.0 core:save (input.hidden-input)
     -1:57.1.0 core:confirm (input.hidden-input)
  4x -1:46.3.0 core:move-down (input.hidden-input)
  2x -1:45 editor:newline (input.hidden-input)
  2x -1:42.5.0 core:backspace (input.hidden-input)
     -1:33.7.0 editor:newline (input.hidden-input)
     -1:26.9.0 core:backspace (input.hidden-input)
     -1:24.3.0 editor:newline (input.hidden-input)
  4x -1:14.7.0 core:backspace (input.hidden-input)
     -1:13.1.0 editor:newline (input.hidden-input)
     -1:05.6.0 core:save (input.hidden-input)
     -0:13.1.0 glsl-preview:toggle (div.tool-panel.tree-view)
```

### Non-Core Packages

```
advanced-open-file 0.16.6 
atom-beautify 0.30.5 
atom-material-syntax 1.0.6 
atom-material-ui 2.0.4 
atom-monokai 0.10.9 
atom-yeoman 0.3.15 
autocomplete-clang 0.11.3 
autocomplete-glsl 0.2.3 
busy-signal 1.4.3 
clang-format 1.29.0 
glsl-preview 1.2.0 
go-debug 1.5.0 
go-plus 5.5.5 
go-signature-statusbar 1.2.3 
intentions 1.1.5 
language-common-lisp 0.9.5 
language-glsl 2.0.1 
language-lisp 0.2.0 
linter 2.2.0 
linter-clang 4.1.2 
linter-clojure 1.2.0 
linter-glsl 2.1.3 
linter-ui-default 1.6.8 
lisp-paredit 0.5.5 
monokai 0.24.0 
parinfer 1.20.0 
rainbow-delimiters 1.3.0 
rainbow-tabs 2.5.1 
seti-ui 1.9.0 
tabs-to-spaces 1.0.3 
xml-formatter 0.11.0 
```

**Comment by mysteryDate** [comment]
Seeing this issue when copy-pasting from shadertoy. It seems to away when changing `fragColor` to `gl_FragColor`, but I can't re-create it consistently.

  @fordhurley Should we bind the keyword `fragColor` to `gl_FragColor` like with do with `u_resolution` etc.? Or should we not let that kinda stuff get out of hand?

**Comment by mysteryDate** [comment]
Ah, nevermind. That's definitely not the issue.

**Comment by mysteryDate** [comment]
Ah! It's not `fragColor` vs `gl_FragColor`, it's `main` vs `mainImage`!

Why the hell does shadertoy need to rename everything?

**Comment by mysteryDate** [comment]
That's definitely the problem. 

<img width="991" alt="screen shot 2017-09-01 at 15 41 09" src="https://user-images.githubusercontent.com/2529989/29985219-0358f410-8f2c-11e7-8ec5-8f44c8257f3d.png">

We kinda threw the wrong error, fwiw.

**Comment by fordhurley** [comment]
Thanks @Piot for @mysteryDate for the detective work. Looks like we needed to be a little more cautious when parsing out the line number from the shader error message, because there's no actual line to call "the error" when main is missing. After e6de627, we get an error message like:

<img src="https://user-images.githubusercontent.com/200646/30038913-2c5e051c-9198-11e7-8f61-af94da9ca593.png" width="400">

**Comment by fordhurley** [comment]
The fix is published in the v1.2.1 release.
